atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -650,13 +660,158 @@ def set_build_image(self):
 
         if build_kind == 'DockerImage':
-            self.config_kwargs['build_image'] = build_name
+            return build_name
         else:
             raise RuntimeError("Build kind isn't 'DockerImage' but %s" % build_kind)
 
+    def set_build_image(self, platforms):
+        """
+        Overrides build_image for worker, to be same as in orchestrator build
+        """
+        orchestrator_plat = 'x86_64'
+        status = get_build_json().get("status")
+        build_config_name = None
+        build_image = None
+        imagestream = None
+        need_manifest = not (orchestrator_plat in platforms and len(platforms) == 1)
+        use_same_image = False
+        current_buildimage = self.get_current_buildimage()
+        self.plat_digests = {}
+
+        # BuildConfig exists
+        if 'config' in status:
+            config = status['config']
+            if config['kind'] == 'BuildConfig':
+                build_config_name = config['name']
+            else:
+                raise RuntimeError("Build config type isn't BuildConfig : %s" % config['kind'])
+
+        # get image build from build metadata, which is set for direct builds
+        else:
+            metadata = get_build_json().get("metadata")
+
+            if 'from' in metadata['annotations']:
+                scratch_from = json.loads(metadata['annotations']['from'])
+
+                if scratch_from['kind'] == 'DockerImage':
+                    build_image = scratch_from['name']
+                elif scratch_from['kind'] == 'ImageStreamTag':
+                    imagestream = scratch_from['name']
+                else:
+                    raise RuntimeError("Build annotation has unknown 'kind' %s" %
+                                       scratch_from['kind'])
+            elif not need_manifest:
+                use_same_image = True
+            else:
+                raise RuntimeError("Build wasn't created from BuildConfig and neither"
+                                   "has 'from' annotation, which is needed for specified arch")
+
+        osbs = get_openshift_session(self.workflow, self.openshift_fallback)
+
+        # inspect BuildConfig
+        if build_config_name:
+            try:
+                build_config = osbs.os.get_build_config(build_config_name)
+            except OsbsException:
+                raise RuntimeError("Build config not found : %s" % build_config_name)
+
+            try:
+                build_name = build_config['spec']['strategy']['customStrategy']['from']['name']
+                build_kind = build_config['spec']['strategy']['customStrategy']['from']['kind']
+            except KeyError:
+                raise RuntimeError("BuildConfig object is malformed")
+
+            if build_kind == 'DockerImage':
+                build_image = build_name
+            elif build_kind == 'ImageStreamTag':
+                imagestream = build_name
```

Comment:
Can some of this be factored out to avoid repeating it? e.g. `build_image, imagestream = process_from(...)`.

Reply:
ok